src/libPMacc/include/simulationControl/SimulationHelper.hpp

```diff
@@ -135,7 +143,11 @@ class SimulationHelper : public IPlugin
             Environment<DIM>::get().PluginConnector().checkpointPlugins(currentStep,
                                                                         checkpointDirectory);
 
-            GridController<DIM> &gc = Environment<DIM>::get().GridController();
+            /* important synchronize: only if no errors occured until this
+             * point guarantees that a checkpoint is usable */
+            CUDA_CHECK(cudaDeviceSynchronize());
+            CUDA_CHECK(cudaGetLastError());
+
             MPI_CHECK(MPI_Barrier(gc.getCommunicator().getMPIComm()));
```

Comment:
ah, these barriers will kill us in traces ;)

we need a `MPI_IBarrier` (MPI 3+) at this point :)